src/domain/users/users.repository.ts

```diff
@@ -46,4 +51,42 @@ export class UsersRepository implements IUsersRepository {
       },
     );
   }
+
+  async addWalletToUser(args: {
+    newSignerAddress: `0x${string}`;
+    authPayload: AuthPayload;
+  }): Promise<Pick<Wallet, 'id'>> {
+    if (!args.authPayload.signer_address) {
+      throw new UnauthorizedException();
+    }
+    return await this.postgresDatabaseService.transaction(
+      async (entityManager: EntityManager) => {
+        const walletRepository = entityManager.getRepository(Wallet);
+
+        const authenticatedWallet = await walletRepository.findOne({
+          where: { address: args.authPayload.signer_address },
+          relations: { user: true },
+        });
+
+        if (!authenticatedWallet?.user) {
+          throw new NotFoundException('User not found');
+        }
+
+        try {
+          const walletInsertResult = await walletRepository.insert({
+            user: authenticatedWallet.user,
+            address: args.newSignerAddress,
+          });
+          return { id: walletInsertResult.identifiers[0].id };
+        } catch (error) {
+          if (error instanceof QueryFailedError) {
+            throw new ConflictException(
+              `A wallet with the same address already exists`,
+            );
+          }
```

Comment:
We are assuming that every query failure denotes a unique violation. We should also check the error code, for which [unique violation is 23505](https://www.postgresql.org/docs/current/errcodes-appendix.html).